app/src/main/java/com/sun/makeupwindow/data/source/local/dao/FavoriteProductDaoImpl.kt

```diff
@@ -0,0 +1,63 @@
+package com.sun.makeupwindow.data.source.local.dao
+
+import android.content.ContentValues
+import com.sun.makeupwindow.data.model.Product
+import com.sun.makeupwindow.data.source.local.db.AppDatabase
+
+class FavoriteProductDaoImpl private constructor(
+    private val appDatabase: AppDatabase
+) : FavoriteProductDao {
+
+    private val writableDatabase = appDatabase.writableDatabase
+    private val readableDatabase = appDatabase.readableDatabase
+
+    override fun getFavoriteProduct(): List<Product> {
+        val listProduct = mutableListOf<Product>()
+        val query =
+            "SELECT * FROM ${Product.TABLE_FAVORITE_NAME} INNER JOIN ${Product.TABLE_NAME} " +
+                    "ON ${Product.TABLE_FAVORITE_NAME}.${Product.ID_FAVORITE_PRODUCT} = " +
+                    "${Product.TABLE_NAME}.${Product.ID};"
+        val cursor = readableDatabase.rawQuery(query, null)
+        cursor.use {
+            it.moveToFirst()
+            while (it.isAfterLast) {
+                val listColor = ColorDaoImpl.getInstance(appDatabase)
+                    .getColor(it.getInt(it.getColumnIndex(Product.TABLE_NAME + "." + Product.ID)))
+                listProduct.add(Product(it, listColor))
+                it.moveToNext()
+            }
+        }
+        return listProduct
+    }
+
+    override fun insertFavoriteProduct(idProduct: Int): Boolean {
+        val value = ContentValues().apply {
+            put(Product.ID_FAVORITE_PRODUCT, idProduct)
+        }
+        return writableDatabase.insert(
+            Product.TABLE_FAVORITE_NAME,
+            null,
+            value
+        ) > 0
```

Comment:
có thể viết thành đc 1 dòng như thế này ko?
```suggestion
override fun inserProduct(product: Product) =
   writableDatabase.insert(Product.TABLE_FAVORITE_NAME, null, product.getContentValue())
```

Reply:
em cx đã nghĩ đến như này rồi ạ, nhưng trong class data Product đã có getContentValue() rồi ạ